### ISSUE TITLE
Provide a method to reverse the argument list for function?

### ISSUE THREAD

**Issue by yankaifyyy** [evhub/coconut#170]
When the configurable parameters of a function don't appear in the front of the param list, the higher-order function may be meaningless.

For example, a function to check integer must be

``` python
def is_integer(n) = n `isinstance` int
```

But in a functional fashion

``` python
is_integer = isinstance$(int)
```

may be better.

So if there is any way to reverse the argument list, so that the currying makes sense?

Thanks a lot

**Comment by evhub** [comment]
I could add this as a built-in, although you should just be able to do
```
def swap2(f) = (y, x) -> f(x, y)
```
or for an arbitrary number of arguments
```
def swap(f) = (*args) -> f(*reversed(args))
```

**Comment by darkf** [comment]
`flip` (`flip f x y = f y x`) is a nice built-in.

**Comment by evhub** [comment]
Closing this as won't fix in favor of #201.
